cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/async/AsyncKafkaCruiseControl.java

```diff
@@ -60,9 +63,17 @@ public AsyncKafkaCruiseControl(KafkaCruiseControlConfig config, MetricRegistry d
    * @see KafkaCruiseControl#state(OperationProgress)
    */
   public OperationFuture<KafkaCruiseControlState> state() {
+    // All substates are requested by default.
```

Comment:
currently the only place we call this old `state()` is in `getGoalsAndRequirements()` and we only use monitor and analyzer state there. maybe we can change the invocation there to not get executor state, or not since getting executor state is computationally cheap.